lib/competition_info.rb

```diff
@@ -6,6 +6,7 @@ class CompetitionInfo
   def initialize(competition_json)
     @competition_json = competition_json
     @competition_id = competition_json['id']
+    @qualifications = fetch_qualifications
```

Comment:
Hm, not super happy about making a network call in a constructor.
Are there any other approaches? Why do we even need to hold/cache the `qualifications` data as part of `competition_info`?

Either
- pass in the already fetched thing into the constructor or
- don't have `qualifications` be part of the `competition_info` model and fetch the data from the network on-demand, ie only in places where it's actually needed.

Reply:
My thinking on this is basically: All competition-related information is logically related, and it makes ergonomic and practical* sense to have it be one object, and I want to abstract away the implementation detail of how a comp's qualification data gets sourced from our business logic. 

If we passed the already-fetched thing into the constructor, I still feel like I wouldn't want to be fetching it in our business logic because that just clogs things up further.

_*because we require the qualification info anywhere that we're using a CompetitionInfo object, at least at the moment._



At the moment the qualification data is needed in all places where CompetitionInfo is needed. I think having them as different endpoints makes sense in the monolith, but for our use case in wca-reg, it's much more ergonomic (in my mind at least) to have it all part of one object.